### ISSUE TITLE
Spago Support

### ISSUE THREAD

**Issue by danieljharvey** [ethul/purs-loader#125]
Hello!

Hope you are well! I have gotten `purs-loader` working nicely with [Spago](https://github.com/spacchetti/spago) with the following webpack config code - and wondered if this is the kind of thing that would make sense to include in `purs-loader` (with an `includeSpagoSources` flag or something).

I previously tried just using a glob but doing so tries to compile example code that may not have the dependencies it requires - hence this code calls `spago sources` each time to get an updated list of the correct folders.

What do you think? If it makes sense I'll submit a PR.

```
const { execSync } = require('child_process');
const os = require('os');

const getSpagoSources = function() {
  return execSync('spago sources').toString().split(os.EOL).filter(a => a.length > 3)
}

const spagoSources = getSpagoSources()
```

```
{
        test: /\.purs$/,
        exclude: /node_modules/,
        loader: 'purs-loader',
        options: {
          src: [
            ...spagoSources,
            'src/**/*.purs'
          ],
          pscIde: true
        }
      }
```

(disclaimer - I am sure the way I am calling out to shell is hacky as hell, my Node is a bit rusty and I would welcome a more elegant way to go about this)

**Comment by ethul** [comment]
Thanks for working on this. I think it would be nice to support Spago sources. Good idea!

I wonder if this can work much the same way that the loader supports psc-package sources:
https://github.com/ethul/purs-loader/blob/bf7d67bb123db582285fe0c7930dda56737b5701/src/index.js#L62-L92

If I might make a suggestion, maybe the option can just be called `spago`. Currently, we have `pscPackage` as an option, so I think this would work. What do you think?

**Comment by danieljharvey** [comment]
Yeah - `spago` is probably more sensible as a name to keep things consistent. Looks like you've got a slightly less awful method for shelling out so will shamelessly copy this as well.

**Comment by ethul** [comment]
Sounds good. Thanks for this update.

**Comment by ethul** [comment]
Closed by #126
